torch_xla/csrc/device.h

```diff
@@ -15,15 +15,27 @@ namespace torch_xla {
 // TODO(yeounoh) `SPMD` is a virtual device that defers data `TransferToDevice`
 // until after the paritioning pass. This avoids transfering  the full input
 // tensor to the device.
-enum class XlaDeviceType { CPU, CUDA, ROCM, GPU, TPU, XPU, NEURON, SPMD };
+enum class XlaDeviceType { CPU, CUDA, TPU, NEURON, SPMD, PLUGIN };
 
 struct DeviceType : public torch::lazy::BackendDeviceType {
-  DeviceType() { type = static_cast<int>(XlaDeviceType::CPU); }
-  DeviceType(XlaDeviceType xla_device_type) {
-    type = static_cast<int>(xla_device_type);
-  }
+  DeviceType(XlaDeviceType xla_device_type)
+      : torch::lazy::BackendDeviceType(static_cast<int>(xla_device_type)),
+        type_name_(XlaDeviceTypeToString(xla_device_type)) {}
+  DeviceType(const std::string& type_name)
+      : torch::lazy::BackendDeviceType(
+            static_cast<int>(StringToXlaDeviceType(type_name))),
+        type_name_(type_name) {}
+
+  // TODO(wcromar): do we even want this default constructor?
+  DeviceType() : DeviceType(XlaDeviceType::CPU){};
```

Comment:
I felt like we should eventually let PyTorch/XLA to auto detect the device based on some rules(libtpu, cuda version etc). Can't really think of a case we want to default construct a CPU device...

Reply:
I'm 90% sure this constructor is never called. Logically, the "default" device type is managed by `xla_backend_impl`.

Reply:
Actually, we definitely don't need this. Upstream provides a better default constructor: https://github.com/pytorch/pytorch/blob/16d69290c6d037a25e32220b9517597d04dbd0bf/torch/csrc/lazy/backend/backend_device.cpp#L14-L15

I'm just going to delete this now instead of leaving the TODO.